app/lib/maat/decision.rb

```diff
@@ -2,11 +2,18 @@ module Maat
   class Decision < Dry::Struct
     attribute :reference, Types::Integer.optional
     attribute :maat_id, Types::Integer
-    attribute :interests_of_justice, Types::InterestsOfJusticeDecision
-    attribute :means, Types::MeansDecision
-    attribute :funding_decision, Types::FundingDecisionResult
+    attribute? :case_id, Types::String.optional
+    attribute :interests_of_justice, Types::InterestsOfJusticeDecision.optional
+    attribute :means, Types::MeansDecision.optional
+    attribute :funding_decision, Types::FundingDecisionResult.optional
     attribute? :comment, Types::String.optional
 
+    def checksum
+      Digest::MD5.hexdigest(to_json)
+    end
```

Comment:
Maybe a silly question but why is this required?

Reply:
We store this in Review and use it to find out if the decision on MAAT has changed since we last updated the Review record here:

https://github.com/ministryofjustice/laa-review-criminal-legal-aid/pull/726/files#diff-2e01c001095fbec6bcf1621ac7e2de1b89af757b5f2e2b88549688e6ba5bf725R32